<?xml version="1.0"?>
  <launch>
    <arg name="voxel_size" default="0.2" />
    <arg name="voxels_per_side" default="16" />
    <arg name="esdf_max_distance_m" default="2.0"/>
    <arg name="world_frame" default="world" />
    <arg name="clear_radius" default="1.0" />
    <arg name="pointcloud_topic" default="pointcloud" />

      <node name="voxblox_node" pkg="voxblox_ros" type="esdf_server" output="screen" args="-alsologtostderr" clear_params="true">
        <remap from="pointcloud" to="$(arg pointcloud_topic)"/>
        <remap from="voxblox_node/esdf_map_out" to="esdf_map" />
        <remap from="voxblox_node/tsdf_map_out" to="tsdf_map" />

		<param name="autostart" value="true" />
        <!-- Path to save map -->
        <param name="file_path" value="/home/$(env USER)/tmpfs_voxblox" />
        <!-- Publishing settings. -->
        <param name="publish_tsdf_map" value="true" />
        <param name="publish_esdf_map" value="true" />
        <param name="update_mesh_every_n_sec" value="0.25" />
        <param name="update_esdf_every_n_sec" value="0.25" />
        <param name="publish_map_every_n_sec" value="0.25" />
        <!-- New -->
        <param name="publish_slices" value="true" />
        <param name="publish_tsdf_info" value="false" />
        <param name="publish_pointclouds" value="true" />
        <param name="output_mesh_as_pointcloud" value="true" />
        <param name="output_mesh_as_pcl_mesh" value="true" />
        <param name="publish_traversable" value="true" />
        <param name="traversability_radius" value="1.0" />
        <param name="use_const_weight" value="true" />
        <param name="enable_icp" value="false" />
        <!-- -->
        <param name="method" value="fast" />
        <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
        <param name="tsdf_voxels_per_side" value="$(arg voxels_per_side)" />
        <param name="esdf_max_distance_m" value="$(arg esdf_max_distance_m)" />
        <param name="max_ray_length_m" value="10.0" />
        <param name="voxel_carving_enabled" value="true" />
        <param name="color_mode" value="normals" />
        <param name="use_tf_transforms" value="true" />
        <param name="min_time_between_msgs_sec" value="0.10" />
        <param name="clear_sphere_for_planning" value="true" />
        <param name="occupied_sphere_radius" value="4.0" />
        <param name="clear_sphere_radius" value="$(arg clear_radius)" />
        <param name="slice_level" value="1.0" />
        <param name="world_frame" value="$(arg world_frame)" />
        <param name="verbose" value="false" />
        <param name="mesh_filename" value="$(find voxblox_ros)/mesh_results/2.ply" />
      </node>

  </launch>